### PR TITLE
Sp/new stratification, eddy velocity, and diagnostics

### PR DESCRIPTION
**Pull request by Shirui-peng**
A three-layer stratification is added to represent a weakly-stratified mixed layer, a strongly-stratified thermocline, and a moderately stratified deeper layer. The front fields are separated from the eddy fields, and the thermal-wind velocity at the fronts are added as an initial condition. Once we separate the fronts and eddies, the eddy fields have displacement and rotational symmetries between the Lx/2-by-Ly/2 subdomain in the center and other boundary domains. And these symmetries are used to set the eddy velocities and surface levels such that the periodic boundary conditions are satisfied. The cooling is updated to stop after 10 days. And other minor edits are included for some diagnostics.

### CODE REVIEW

File: src/idealized_setup.jl [hunk 1]
```diff
@@ -65,10 +65,11 @@ function idealized_setup(arch;
     
     # # Cooling in the middle of the domain and heating outside?
     # @inline Qtop(x, y, t, p) = - p.Q / p.ρ₀ / p.cₚ * cos(2π * x / p.Lx)
+    @inline Qtop(x, y, t) = Q / ρ₀ / cₚ * (1 - tanh((t - 10days) / 12hours)) / 2
 
     u_top = FluxBoundaryCondition(τw * cosd(θ) / ρ₀)
     v_top = FluxBoundaryCondition(τw * sind(θ) / ρ₀)
-    T_top = FluxBoundaryCondition(Q / ρ₀ / cₚ) # Positive fluxes at the top are cooling in Oceananigans
+    T_top = FluxBoundaryCondition(Qtop)# / ρ₀ / cₚ) # Positive fluxes at the top are cooling in Oceananigans
```
Comment (simone-silvestri) [inline src/idealized_setup.jl:72]: Why changing this? Are you prescribing fluxes of temperature instead of heat?

File: src/idealized_setup.jl [hunk 2]
```diff
@@ -83,7 +84,7 @@ function idealized_setup(arch;
                         settings...)
 
     if isforced(model)
-        set!(model, T = Tᵢ) 
+        set!(model, v = vᶠ, T = Tᵢ) 
     else
         set!(model, u = uᵢ, v = vᵢ, T = Tᵢ)
```
Comment (simone-silvestri) [inline src/idealized_setup.jl:89]: If it is not forced we don't need to add this velocity as well?

File: src/idealized_setup.jl (hunk 1 above)
Comment (Shirui-peng) [reply src/idealized_setup.jl:72]: It's testing a time dependent temperature flux Qtop defined above to stop the flux after 10 days.

File: src/idealized_setup.jl (hunk 2 above)
Comment (Shirui-peng) [reply src/idealized_setup.jl:89]: Nice catch! I will add this as well.

**Review by simone-silvestri — APPROVED** [review]

**Pull request merged**